compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -515,11 +515,7 @@ impl<'f, 'tcx> Coerce<'f, 'tcx> {
     ///
     /// [unsized coercion](https://doc.rust-lang.org/reference/type-coercions.html#unsized-coercions)
     #[instrument(skip(self), level = "debug")]
-    fn coerce_unsized(&self, mut source: Ty<'tcx>, mut target: Ty<'tcx>) -> CoerceResult<'tcx> {
-        source = self.shallow_resolve(source);
-        target = self.shallow_resolve(target);
```

Comment:
why remove these, are we guaranteed that they are shallow resolved? if so, please convert them into debug asserts

Reply:
yes, we shallowly resolve at the top of the coerce entrypoint